src/neuroconv/tools/nwb_helpers/_configuration_models/_hdf5_dataset_io.py

```diff
@@ -78,3 +80,28 @@ def get_data_io_kwargs(self) -> Dict[str, Any]:
             compression_bundle = dict(compression=self.compression_method, compression_opts=compression_opts)
 
         return dict(chunks=self.chunk_shape, **compression_bundle)
+
+    @classmethod
+    def from_existing_neurodata_object(
```

Comment:
```suggestion
    def from_neurodata_object(
```

IMO don't need the 'existing' part in the name. Evident enough that if it applies to a neurodata object, that object must already exist

However, it is important to note (and test appropriate skip conditions) the application to neurodata objects that are in-memory instead of written/read to/from disk, which can change the way a couple of things behave